packages/web-extension/src/walletManager/util.ts

```diff
@@ -7,6 +10,19 @@ export const walletChannel = (walletName: WalletManagerProps['walletName']) =>
 
 export const walletManagerProperties: RemoteApiProperties<WalletManagerApi> = {
   activate: RemoteApiPropertyType.MethodReturningPromise,
-  clearStore: RemoteApiPropertyType.MethodReturningPromise,
-  deactivate: RemoteApiPropertyType.MethodReturningPromise
+  deactivate: RemoteApiPropertyType.MethodReturningPromise,
+  destroy: RemoteApiPropertyType.MethodReturningPromise
+};
+
+/**
+ * Compute a unique walletId from the keyAgent chainId and the root public key hash.
+ * `networkId-networkMagic-blake2bHashOfExtendedAccountPublicKey`
+ */
+export const getWalletId = async (keyAgent: AsyncKeyAgent): Promise<string> => {
+  const { networkId, networkMagic } = await keyAgent.getChainId();
+  const extendedAccountPublicKey = await keyAgent.extendedAccountPublicKey();
+  const pubKey = Buffer.from(extendedAccountPublicKey, 'hex');
+  const pubKeyHash = blake2b(64).update(pubKey).digest('hex');
+
+  return `${networkId}-${networkMagic}-${pubKeyHash}`;
```

Comment:
We can discuss whether this is the best way to build the ID. It is kind of lengthy, and the `extendedAccountPublicKey` and its hash have the same length

Reply:
It's semantically nice. If length is an issue we could truncate the public key hash to 7 characters or so, but probably best to keep it in original form until there's a reason to think about shortening.

Reply:
Done: [7fb0dd5](https://github.com/input-output-hk/cardano-js-sdk/pull/560/commits/7fb0dd528e1c2d3b9d0eb4556d3f7e35ffe08f48)